### ISSUE TITLE
StdLib Man: Fix Chapters' Anchor IDs

### ISSUE THREAD

**Issue by tajmone** [AnssiR66/AlanStdLib#81]
- [ ] Shorten long chapter titles.
- [ ] Use title-based anchor IDs instead of chapter number.

Currently, each chapter of the StdLib Man has been assigned an anchor ID based on its chapter number (`ch1`, `ch2`, etc.). There are some downsides with this approach:

1. If we add/delete chapters, of if we move them around, the anchors need to renamed.
2. Their XRfes don't provide meaningful hints in the sources — e.g. `see <<ch10>> for more info` (Which chapter is that? one needs to look it up).

It would be much better to give to each chapter an anchor ID which resembles its real title. E.g.:

- "Ch. 1. Introduction" &rarr; `ch.intro`
- "Ch. 10. The my_game instance and its attributes" &rarr; `ch.my_game`
- "Ch. 12. Runtime messages" &rarr; `ch.runtime-messages`

and so on.

This way, even we add new chapters, or change their order, we won't need to change their anchor IDs; and in the source file XRefs will be intuitive to understand: `see <<ch.my_game>> for more info` (we immediately know which chapter is being referenced here).

The main obstacle right now is that there are many titles which are too long ([see comments in #76]) to be resumed in a meaningful and resemblant anchor ID — e.g.:

- "Ch 7. Additional attributes for THINGs: (NOT) distant, (NOT) reachable, scenery"

It would be hard to create a short anchor ID that would intuitively represent that title — `ch.things-attributes`, `ch.additional-things-attributes`, `ch.things-additional`? None of these would be intuitively associated with the chapter, nor easy to memorize.

Note that these anchor IDs need to be _easy to memorize_ for maintainers, so they don't have to look them up in a reference table each time they need to create an XRef.

[see comments in #76]: https://github.com/AnssiR66/AlanStdLib/issues/76#issuecomment-686607412

**Comment by AnssiR66** [comment]
Ok. Ch.7 could be something like ch.attributes_things?

**Comment by tajmone** [comment]
> Ok. Ch.7 could be something like `ch.attributes_things`?

Yes, as far as the Anchor ID goes that would be a fine solution; the real problem is the current length of the title. Usually a maintainer should be able to intuitively deduce the anchor ID from the full title (i.e. it's just a shortened version, dropping articles, particles, etc.).

The fact that one can't currently deduce the Anchor ID from some of these titles it's just an indicator that they are too long.

In reality, these titles weren't really titles in the original PDF Manual, they were more like normal sentences which were styled strongly to facilitate readers spotting them. But now that we've moved to AsciiDoc format, there's the need to abide to a stricter book format (section titles, stricter section numbering rules, a table of contents, and cross references being managed by Asciidoctor).

What I suggest is adopting a compromise solution for lengthy titles:

1. Shorten the actual chapter/section title
2. Below the title, add a re-adaptation of the original title as a sentence in italics (a sort of resume subtitle, so to speak).

For example, the current title:

> ### 1.1. What to read if you’re a complete newbie

Could become

> ### 1.1. ALAN Newbies
>
> _Where to start if you’re a complete newbie to ALAN IF._

And so on. Basically, sections title will be kept as short as possible, so that readers can spot them rapidly in the TOC, and some sections will have an opening sentence (in italic, maybe slightly bigger font size) describing in more detail what the section is all about. This approach can actually be found in some manuals and reference documentation, and it was used quite a lot in the past, although not as much today; but I personally like it.

IMO, this makes sense for a Manual, for it's intended to read over multiple times, as well as being consulted often for quick solutions — hence, concise titles and linkable cross-references (pointing to exact places, even if in mid-section) will be really appreciated by end users.

**Comment by AnssiR66** [comment]
I think these chapters "complete newbie", "what to read if you have used v1.0" and so on could be omitted altogether, they were more relevant when library 2.0 was new, some 10 years ago. Nowadays we could have just one instruction (if even that) for all users.

**Comment by tajmone** [comment]
> I think these chapters "complete newbie", "what to read if you have used v1.0" and so on could be omitted altogether, they were more relevant when library 2.0 was new, some 10 years ago. Nowadays we could have just one instruction (if even that) for all users.

I thought that too. Maybe the most important aspects today are addressing:

1. ALAN newbies
2. StdLib users migrating from v2.0/2.1 to v2.2 (i.e. "What's new in v2.2")

Chances that there will be users coming from a fresh library v1 experience are quite slim — even if they do, they probably won't have been using ALAN for a decade, at least.

As for point (2), I suggest we'll be adding a "What's New" document (AsciiDoc/HTML) which will contain details of every StdLib update — we'll try our best to cover the updates from 2.0 to 2.1, and then focus on the many new features (and changes) of 2.2.

The Manual itself (IMO) should always focus on providing usage instructions for the current library version. References to the "What's New" document can be dropped here and there, and maybe also the occasional admonition block to highlight recent changes in library usage (e.g. with clothing, which has changed considerably), but these should always be side-notes, except for the "Intro" chapter which might provide some detailed guidance (but referring to other docs as much as possible).
